### DISCUSSION TITLE
Problem using the command line version of Yarle on Ubuntu/WSL

### DISCUSSION THREAD

**Discussion by pmottishaw** [akosbalasko/yarle#291] [Q&A]
Hi, I got the GUI version working - very helpful thanks, but I wanted to process multiple enex files (to get the correct links between all notes) so I installed the command line version on Ubuntu running under WSL on a Windows 10 machine. I installed Node.js v16.5, then executed the command

`npx -p yarle-evernote-to-md@latest yarle --configFile ./config.json`

The first time I ran this I got 
`Need to install the following packages:
  yarle-evernote-to-md@latest
Ok to proceed? (y)`
and I let the install happen.

I tried again with the original command and I got the error message
`sh: 1: yarle: not found`

I have checked carefully to make sure I followed all the instructions

**Comment by akosbalasko** [comment]
Hi @pmottishaw !
Erm... sorry for this long delay, even if I'm subscribed, I don't get notifications about the updates in discussions. 
First of all, I don't think that inter-notebook links are supported at all, I have to test this scenario. It is suspicious that it's not, or at least processing multiple notebooks as a batch is indifferent in this case, because multi notebook processing is done one by one.

**Reply by pmottishaw** [reply]
Thanks for following up. As an example, I have one notebook that produced 88 enex files (there were a lot of large pdf attachements). It is just too many to do "one at a time" with the GUI tool. So even just automating this process would be helpful

However, I found a workaround. I converted the 88 enex files into one 6.7 Gbyte enex file!?

I used Notebook++ to remove the XML code indicating the start and end of the export in each of the enex files except the first and last (a total of 86 enex files). N++ can do this in two passes using "Find in files..." with the appropriate regex. In the first enex I removed the end of export XML code and in the last I removed the start of export XML code. I then merged all 88 text files to produce something that I hoped Yarle would mistake for one very big enex file (it was a 6.7 Gbyte text file). It worked very well!!! It correctly created all but 3 of the notes as Obsidian files with the correct attachements. The 3 notes were skipped - probably because my splicing process had corrupted them. Most links seem to be fine.

Would it be possible to add this type of pre-processing to the Yarle GUI?

With regard to the links; does Yarle just put the link text from Evernote in [[ ]] and hope it is the same as the link title or do you do something more sophisticated?

**Comment by akosbalasko** [comment]
Hi @pmottishaw ,

It looks like a great workaround, but I don't think that it can be integrated into Yarle, because I process the enex file as a stream, otherwise the app would run out of memory in case of large files. 
But I released a new version of Yarle(4.2.0) in where I extended the UI to be able to set multiple enex files to be converted. 

On the links: Yarle tries to recognize the note what the link points to by the title of the link and then use that as you mentioned, with the double brackets. It's not the smartest way, because if there are multple notes with the same name, then the links can be mixed up. Unfortunately the direct guid of a note cannot be exported, I'm still thinking on a better way, probably something with table of contents.

**Comment by pmottishaw** [comment]
Thanks for the new version. I will give it a try.
